kube/src/api/dynamic.rs

```diff
@@ -256,16 +164,129 @@ impl Resource for DynamicObject {
     }
 }
 
+/// Resource scope
+#[derive(Debug, Clone, Hash, Eq, PartialEq)]
+pub enum Scope {
+    /// Objects are global
+    Cluster,
+    /// Each object lives in namespace.
+    Namespaced,
+}
+
+/// Operations that are supported on the resource
+#[derive(Debug, Clone, Hash, Eq, PartialEq)]
+pub struct Operations {
+    /// Object can be created
+    pub create: bool,
+    /// Single object can be queried
+    pub get: bool,
+    /// Multiple objects can be queried
+    pub list: bool,
+    /// A watch can be started
+    pub watch: bool,
+    /// A single object can be deleted
+    pub delete: bool,
+    /// Multiple objects can be deleted
+    pub delete_collection: bool,
+    /// Object can be updated
+    pub update: bool,
+    /// Object can be patched
+    pub patch: bool,
+    /// All other verbs
+    pub other: Vec<String>,
+}
+
+impl Operations {
+    /// Returns empty `Operations`
+    pub fn empty() -> Self {
+        Operations {
+            create: false,
+            get: false,
+            list: false,
+            watch: false,
+            delete: false,
+            delete_collection: false,
+            update: false,
+            patch: false,
+            other: Vec::new(),
+        }
+    }
+}
+/// Contains additional, detailed information abount API resource
+pub struct ApiResourceExtras {
+    /// Scope of the resource
+    pub scope: Scope,
+    /// Available subresources. Please note that returned ApiResources are not
+    /// standalone resources. Their name will be of form `subresource_name`,
+    /// not `resource_name/subresource_name`.
+    /// To work with subresources, use `Request` methods.
+    pub subresources: Vec<(ApiResource, ApiResourceExtras)>,
+    /// Supported operations on this resource
+    pub operations: Operations,
+}
```

Comment:
> Is there particular problems we can solve with this information? 

Well, `dynamic-api` example can skip non-listable resources (instead of ignoring list errors).

As I said, this stuff is the least important part of the PR. It can be easily deleted because anyone who needs this information can get it using `Client` methods and looking at `APIResource` stuff. I don't see a way `kube` itself could benefit from this information.

Reply:
Ah, good point. I was mostly wondering here since it wasn't used. I think it's totally okay to include in the discovery module.

Reply:
Ok. Then I'll move `ApiResourceExtras` (with your suggestions applied) into #491 since it's more about API discovery rather than talking to the api-server.